src/MongoDB.Driver.Core/Core/Authentication/MongoDBX509Authenticator.cs

```diff
@@ -101,18 +111,22 @@ public async Task AuthenticateAsync(IConnection connection, ConnectionDescriptio
         /// <inheritdoc/>
         public BsonDocument CustomizeInitialIsMasterCommand(BsonDocument isMasterCommand)
         {
-            return isMasterCommand; 
+            isMasterCommand.Add("speculativeAuthenticate", CreateAuthenticateCommand());
+            return isMasterCommand;
         }
 
         // private methods
+
+        private BsonDocument CreateAuthenticateCommand() => new BsonDocument
+        {
+            { "authenticate", 1 },
+            { "mechanism", Name },
+            { "user", _username, _username != null }
+        };
```

Comment:
In general I would prefer expression bodied methods to be limited to one-liners.

Specially bad in this case is that the way the opening and closing braces are formatted makes this "look" like a traditional method.

If you really prefer an expression bodied method even for something that can't fit in one line at least format it to make it visually obvious that this method is an expression bodied method and not a traditional method:

    private BsonDocument CreateAuthenticateCommand()
        =>
            new BsonDocument
            {
                { "authenticate", 1 },
                { "mechanism", Name },
                { "user", _username, _username != null }
            };

Reply:
I've reformatted this the "traditional" way :)